### PR TITLE
Add "Install Kyma with custom Istio deployment" docs

### PR DESCRIPTION
**Pull request by tomekpapiernik**
**Description**

Changes proposed in this pull request:

- Added an "installation" type document describing how to install Kyma with custom Istio deployment.
- Updated Istio installation description document in the "Service Mesh" topic. 

**Related issue(s)**
See #704

### CODE REVIEW

File: docs/kyma/docs/037-inst-custom-istio.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+You can use Kyma with a custom deployment of Istio that you installed in the target environment. To enable such implementation, remove Istio from the list of components that install with Kyma.
+The version of your Istio deployment must match the version that Kyma currently supports.
+
+In the installation process, the installer applies a custom patch to every Istio deployment. This is a mandatory step.  
+
+>**NOTE:** To learn more, read the **Istio patch** document in the **Service Mesh** documentation topic.
+
+## Prerequisites
+
+- A live Istio version compatible with the version currently supported by Kyma. To check the supported version, see the `resources/istio-kyma-patch/templates/job.yaml` file.
```
Comment (pbochynski) [inline docs/kyma/docs/037-inst-custom-istio.md]: Please provide a link to istio installation guide.

Comment (tomekpapiernik) [reply docs/kyma/docs/037-inst-custom-istio.md]: Added

File: docs/kyma/docs/037-inst-custom-istio.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+title: Installation with custom Istio deployment
+type: Installation
+---
+
+You can use Kyma with a custom deployment of Istio that you installed in the target environment. To enable such implementation, remove Istio from the list of components that install with Kyma.
```
[the remark this answers is not in the document]
Comment (tomekpapiernik) [reply docs/kyma/docs/037-inst-custom-istio.md]: Yes, it's "some environment", not an Environment in Kyma. 

I imply future action - you edit the list before the installation and it has direct effect on what is installed.

[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/kyma/docs/037-inst-custom-istio.md]: 👍

**Pull request merged**